src/approaches/open_loop_llm_approach.py

```diff
@@ -75,40 +100,22 @@ def _predict(self, state: State, atoms: Set[GroundAtom],
             temperature=CFG.open_loop_llm_temperature,
             seed=CFG.seed,
             num_completions=CFG.open_loop_llm_num_completions)
-        # Try to convert the output into an abstract plan.
-        for llm_prediction in llm_predictions:
-            ground_nsrt_plan = self._process_single_prediction(
-                llm_prediction, state, atoms, goal)
-            if ground_nsrt_plan is not None:
-                # If valid plan, add plan to memory so it can be refined!
-                memory["abstract_plan"] = ground_nsrt_plan
-                return memory["abstract_plan"].pop(0)
-        # Give up if none of the predictions work out.
-        raise ApproachFailure("No LLM predicted plan achieves the goal.")
+        for pred in llm_predictions:
+            option_plan = self._llm_prediction_to_option_plan(pred, objects)
+            yield option_plan
 
-    def _process_single_prediction(
-            self, llm_prediction: str, state: State, atoms: Set[GroundAtom],
+    def _option_plan_to_nsrt_plan(
+            self, option_plan: List[Tuple[ParameterizedOption,
+                                          Sequence[Object]]],
+            atoms: Set[GroundAtom], objects: Set[Object],
```

Comment:
can we rename atoms to init_atoms here?

Reply:
actually i think that's wrong... the use of `atoms` here comes from the `predict` abstract method in the nsrt_metacontroller_approach, which is run on every step, not just on the init_atoms